locations/templates/locations/detail.html

```diff
@@ -33,21 +34,92 @@ <h3 class="subtitle has-text-centered">Description</h3>
                     <p>{{ location.description|markdownify }}</p>
                 </div>
             </div>
-            <div class="column content image has-background-light has-text-centered">
-                <h3 class="subtitle">Map</h3>
-                <a href="{{ location.map.url }}">
-                    <img class="is-rounded map is-inline-block" alt="A Map of {{ location.name }}"
-                         src="{{ location.map.url }}"/>
-                </a>
-            </div>
+            {% if location.map %}
+                <div class="column content image has-background-light has-text-centered">
+                    <h3 class="subtitle">Map</h3>
+                    <a href="{{ location.map.url }}">
+                        <img class="is-rounded map is-inline-block" alt="A Map of {{ location.name }}"
+                             src="{{ location.map.url }}"/>
+                    </a>
+                </div>
+            {% endif %}
         </div>
-        <div class="columns">
-            <div class="column content has-background-light">
-                <h3 class="subtitle has-text-centered">History</h3>
-                <p>{{ location.history|markdownify }}</p>
+        {% if location.history %}
+            <div class="columns">
+                <div class="column">
+                    <div class="content has-background-light">
+                        <h3 class="subtitle has-text-centered">History</h3>
+                        <p>{{ location.history|markdownify }}</p>
+                    </div>
+                </div>
             </div>
-        </div>
+        {% endif %}
+        {% if location.racedistribution_set.exists %}
+            <div class="columns">
+                <div class="column">
+                    <div class="content has-background-light has-text-centered">
+                        <h3 class="subtitle">Occupant Distribution</h3>
+                        <canvas style="max-width:600px;max-height: 400px;" id="occupancyChart" width="600"
+                                height="400" class="is-inline-block"></canvas>
+                        <br/>
+                        <button class="button is-info" id="button-random-occupant">Random Occupant</button>
+                        <p id="p-random-occupant"></p>
+                    </div>
+                </div>
+            </div>
+        {% endif %}
     </div>
+{% endblock %}
+
+{% block extra_media %}
+    {% if location.racedistribution_set.exists %}
+        <script src="{% static 'node_modules/chart.js/dist/Chart.min.js' %}"></script>
+        <script src="{% static 'node_modules/chartjs-plugin-colorschemes/dist/chartjs-plugin-colorschemes.min.js' %}"></script>
+        <script>
+            let ctx = document.getElementById("occupancyChart").getContext('2d');
```

Comment:
We can use jQuery here for consistency, too:
```suggestion
            let ctx = $("#occupancyChart").first().getContext('2d');
```

Reply:
For some reason .first() won't do the job. .get(0) seems to work though, thanks!

Reply:
Good catch :) `first()` returns a jQuery object while `get()` returns an Element.
`get()` is the correct way